src/main/scala/com/gu/mobile/logback/MobileLogstashEncoder.scala

```diff
@@ -18,14 +19,14 @@ final class MobileLogstashEncoder extends LogstashEncoder {
          |  <defaultAppName>APP_NAME_HERE</defaultAppName>
          |</encoder>
       """.stripMargin))
-    JsObject((AppIdentity.whoAmI(defaultAppName = defaultAppName) match {
+    JsObject(((AppIdentity.whoAmI(defaultAppName = defaultAppName) match {
       case awsIdentity: AwsIdentity => Map(
         "app" -> awsIdentity.app,
         "stack" -> awsIdentity.stack,
         "stage" -> awsIdentity.stage
       )
       case _ => Map("app" -> defaultAppName)
-    }).mapValues(JsString))
+    }) ++ Option(EC2MetadataUtils.getInstanceId).map("ec2_instance" -> _)).mapValues(JsString))
```

Comment:
I don't know if that method is cached, would it be safer to have a private attribute to store the instance ID at the class level, and re-use that attribute upon later calls?

I'm worried this actually does some sort of processing each time

Reply:
:+1:

Reply:
whoAmI already calls getInstanceId so this problem would already exist.

However, to be safe, I've turned the whole app stack stage object into a lazy val